### DISCUSSION TITLE
Système de plugin dans mercator ?

### DISCUSSION THREAD

**Discussion by Arkhee** [dbarzin/mercator#254] [Q&A]
Bonjour,
Nous envisageons le développement d'une micro-application dont l'objet serait de transférer des données depuis un GLPI.
La question se pose sur la forme donnée à cette application : soit une application indépendante qui serait située entre le GLPI et Mercator, soit un développement sous la forme d'un module rattaché à l'une ou l'autre de ces applications.
Existe-t-il un système de plugin dans mercator, qui permettrait d'intégrer cette application sans toucher le coeur de Mercator ?
Autre question, un MCD, par ex. au format mysqlworkbench ou autre, est-il disponible ?
Merci d'avance

**Comment by dbarzin** [comment]
Bonjour,
Très content d'apprendre cette intégration de Mercator avec un GLPI !
Il existe une API dans Mercator qui permet d'alimenter la base de données. Cette API est [documentée](https://dbarzin.github.io/mercator/api/). Il existe des exemples d'utilisation de l'API avec PHP, Python et Bash. Il reste probablement des bugs dans l'API car elle est peu utilisée.
Le modèle de données est également [documenté](https://dbarzin.github.io/mercator/model/).  Bon travail !
Cordialement,
Didier

**Comment by Arkhee** [comment]
Bonjour et merci pour votre retour rapide,
J'ai fait quelques tests avec l'API REST, je ne vois pas de blocage de ce
côté pour le moment. En revanche la question se pose de l'emplacement de
cette fonction de transfert : il s'agira d'une application web, mais la
question est "comment y accéder ?"
Idéalement il pourrait y avoir un menu dans mercator qui ferait apparaître
le formulaire de contrôle, ex. dans "Vues => Importer les données de GLPI",
mais il me semble qu'en l'état actuel, i.e. en l'absence d'un système de
plugin dans mercator, cela nécessiterait une modification du coeur de
l'application ?
Dans notre situation le GLPI n'est pas modifiable donc créer un plugin pour
ce dernier n'est pas envisageable.
En l'absence de solution côté Mercator je réaliserai le développement comme
un site indépendant, mais même dans ce cas il serait pratique de pouvoir
mettre un lien vers ce dernier dans Mercator.
Cordialement

Le mar. 8 nov. 2022 à 17:55, Didier Barzin ***@***.***> a
écrit :

**Comment by dbarzin** [comment]
Bonjour,
Vous pouvez faire une petite application Web hébergée sur le même serveur que Mercator qui remplira cette fonction d'alimentation de la cartographie. Modifier l'application risque de poser des difficultés de mise à jour de celle-ci par la suite.
Cordialement,
Didier
